From a3c41814b07c7c4191a7b369ea3ddf59fb2c23a9 Mon Sep 17 00:00:00 2001
From: Jaeyoon Jung <jaeyoon.jung@lge.com>
Date: Fri, 23 Feb 2018 08:56:31 +0900
Subject: [PATCH] Fix baseUrl returning an empty url

Do not append the directory separator if the current path is the root
directory. Otherwise it is treated as a remote path incorrectly and an
empty url is returned as a result.

Change-Id: Icc62114702265e9e1b247624370a78999e564d5f
Upstream-Status: Submitted [https://codereview.qt-project.org/221224]
---
 src/qml/qml/qqmlengine.cpp | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/qml/qml/qqmlengine.cpp b/src/qml/qml/qqmlengine.cpp
index a32a13b5a4..744b853fa9 100644
--- a/src/qml/qml/qqmlengine.cpp
+++ b/src/qml/qml/qqmlengine.cpp
@@ -1341,7 +1341,10 @@ QUrl QQmlEngine::baseUrl() const
 {
     Q_D(const QQmlEngine);
     if (d->baseUrl.isEmpty()) {
-        return QUrl::fromLocalFile(QDir::currentPath() + QDir::separator());
+        if (QDir::currentPath() == QDir::rootPath())
+            return QUrl::fromLocalFile(QDir::rootPath());
+        else
+            return QUrl::fromLocalFile(QDir::currentPath() + QDir::separator());
     } else {
         return d->baseUrl;
     }
